src/lib/gadgets/bitwise.ts

```diff
@@ -11,7 +11,45 @@ import {
 } from './common.js';
 import { rangeCheck64 } from './range-check.js';
 
-export { xor, and, rotate };
+export { xor, not, and, rotate };
+
+function not(a: Field, length: number, checked: boolean = false) {
+  // check that input length is positive
+  assert(length > 0, `Input length needs to be positive values.`);
+
+  // Check that length does not exceed maximum field size in bits
+  assert(
+    length <= Field.sizeInBits(),
+    `Length ${length} exceeds maximum of ${Field.sizeInBits()} bits.`
+  );
+
+  // obtain pad length until the length is a multiple of 16 for n-bit length lookup table
+  let padLength = Math.ceil(length / 16) * 16;
+
+  // handle constant case
+  if (a.isConstant()) {
+    let max = 1n << BigInt(padLength);
+    assert(
+      a.toBigInt() < max,
+      `${a.toBigInt()} does not fit into ${padLength} bits`
+    );
+    return new Field(Fp.not(a.toBigInt(), length));
+  }
+
+  // create a bitmask with all ones
+  let allOnesF = new Field(2n ** BigInt(length) - 1n);
+
+  let allOnes = Provable.witness(Field, () => {
+    return allOnesF;
+  });
+
+  allOnesF.assertEquals(allOnes);
+
+  let notChecked = xor(a, allOnes, length);
+  let notUnchecked = allOnes.sub(a);
+
+  return checked ? notChecked : notUnchecked;
```

Comment:
we should wrap the actual logic in an if statement, rather than the return values using a ternary operation

```ts
if(checked) {
   // do checked, return checked
} else {
   // do unchecked, return unchecked 
}

```